module-new/src/components/toast/toast.stories.tsx

```diff
@@ -15,28 +16,76 @@ export default {
 } as Meta<typeof Toast>;
 
 /** component template */
-
-const demoToast: IToast = { title: 'Here is a toast!', description: 'Here is the description for my toast' };
-
 const Template: StoryObj<typeof Toast> = {
+  args: {
+    title: 'Here is a toast!',
+    description: 'Here is the description for my toast',
+  },
   decorators: [
     Story => (
       <ArmstrongProvider>
         <Story />
       </ArmstrongProvider>
     ),
   ],
-  render: () => {
+  render: args => {
     const dispatch = useToast();
 
-    return <Button onClick={() => dispatch(demoToast)}>Send a toast</Button>;
+    return <Button onClick={() => dispatch(args)}>Send a toast</Button>;
   },
 };
 
 /** stories */
 
 export const Default: StoryObj<typeof Toast> = {
   ...Template,
+  play: async ({ canvasElement, args }) => {
+    // launch toast
+    const button = within(canvasElement).getByText('Send a toast');
+    expect(button).toBeVisible();
+    userEvent.click(button);
+
+    // check title and description
+    const title = await findByText(document.body, args.title ?? '');
+    const description = await findByText(document.body, args.description ?? '');
+    expect(title).toBeVisible();
+    expect(description).toBeVisible();
+
+    // check auto-dismiss after default dismiss time of 5000ms + 500ms for animation
+    await waitFor(() => Promise.all([expect(title).not.toBeVisible(), expect(description).not.toBeVisible()]), {
+      timeout: 5500,
+    });
+
+    // launch multiple toasts
+    userEvent.click(button);
+    userEvent.click(button);
+
+    // check close buttons on multi toast
+    const [close1, close2] = await findAllByRole(document.body, 'button', { name: 'Close' });
+    expect(close1).toBeVisible();
+    expect(close2).toBeVisible();
+    userEvent.click(close1);
+    userEvent.click(close2);
+    await waitFor(() => Promise.all([expect(close1).not.toBeVisible(), expect(close2).not.toBeVisible()]));
+  },
+};
+
+export const CustomDuration: StoryObj<typeof Toast> = {
+  ...Template,
+  args: {
+    ...Template.args,
+    duration: 1000,
+  },
+  play: async ({ canvasElement, args }) => {
+    const button = within(canvasElement).getByText('Send a toast');
+    expect(button).toBeVisible();
+    userEvent.click(button);
+
+    const title = await findByText(document.body, args.title ?? '');
+    expect(title).toBeVisible();
+    // check auto-dismiss after custom dismiss time of 1000ms + 500ms for animation
+    await waitFor(() => expect(title).not.toBeVisible(), { timeout: 1500 });
```

Comment:
I would be surprised if testing library didn't include mock timers - that would speed up the test and make it more reliable. Have a look at the docs and see if that's an option?

Reply:
Urgh, yes, having just done a quick read-through it doesn't look like timer mocking is a thing in SB - which makes some sense when I think about it but still urgh.

Think your approach is the sensible one.